### PR TITLE
Pull Request: Keycloak

### PR DESCRIPTION
**Pull request by katsiaryne**

### CODE REVIEW

File: auth-service/src/main/java/com/software/modsen/authservice/controller/AuthController.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+@RequestMapping("/auth")
+@RequiredArgsConstructor
+public class AuthController {
+    private final KeycloakService keycloakUserService;
+    private final UserMapper userMapper;
+    private final UserLoginMapper userLoginMapper;
+
+    @PostMapping("/register")
+    public ResponseEntity<UserResponse> createUser(@RequestBody UserRequest userRequest)
+    {
```
Comment (psidarovich) [inline auth-service/src/main/java/com/software/modsen/authservice/controller/AuthController.java]: Use valid java code convention. Brace should be on the prev line

File: auth-service/src/main/java/com/software/modsen/authservice/dto/request/UserLoginRequest.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+@AllArgsConstructor
+public class UserLoginRequest {
+    private String username;
+    private String password;
+}
```
Comment (psidarovich) [inline auth-service/src/main/java/com/software/modsen/authservice/dto/request/UserLoginRequest.java]: What about immutable class?

Comment (psidarovich) [reply auth-service/src/main/java/com/software/modsen/authservice/dto/request/UserLoginRequest.java]: Here and in other places. It's a good practice to make request|responses immutable

File: auth-service/src/main/java/com/software/modsen/authservice/exception/ServiceUnAvailableException.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package com.software.modsen.authservice.exception;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(HttpStatus.BAD_REQUEST)
```
Comment (psidarovich) [inline auth-service/src/main/java/com/software/modsen/authservice/exception/ServiceUnAvailableException.java]: 502 - BadGateway

**Review by psidarovich — CHANGES_REQUESTED** [review]

File: auth-service/src/main/java/com/software/modsen/authservice/exception/ServiceUnAvailableException.java (hunk 3 above)
Comment (katsiaryne) [reply auth-service/src/main/java/com/software/modsen/authservice/exception/ServiceUnAvailableException.java]: Fixed

File: auth-service/src/main/java/com/software/modsen/authservice/dto/request/UserLoginRequest.java (hunk 2 above)
Comment (katsiaryne) [reply auth-service/src/main/java/com/software/modsen/authservice/dto/request/UserLoginRequest.java]: rewrite on records

File: auth-service/src/main/java/com/software/modsen/authservice/controller/AuthController.java (hunk 1 above)
Comment (katsiaryne) [reply auth-service/src/main/java/com/software/modsen/authservice/controller/AuthController.java]: Fixed

**Review by psidarovich — APPROVED** [review]

**Pull request merged**